src/components/screens/new-questionnaire-screen.tsx

```diff
@@ -7,48 +22,90 @@ import { KeyboardReturn } from "@mui/icons-material";
   const [questionnaireTitle, setQuestionnaireTitle] = useState("");
+  const [questionnaireDescription, setQuestionnaireDescription] = useState("");
+  const [options, setOptions] = useState<Question[]>([]);
+  const [passScoreValue, setPassScoreValue] = useState(0);
 
   const handleQuestionTitleChange = (event: ChangeEvent<HTMLInputElement>) => {
     setQuestionnaireTitle(event.target.value);
   };
 
-  /**
-   * State and functions to handle the questions and options
-   */
-  const [questions, setQuestions] = useState<Question[]>([]);
+  const handleDescriptionChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setQuestionnaireDescription(event.target.value);
+  };
+
+  const handlePassScoreSliderChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setPassScoreValue(Number(event.target.value));
+  };
 
   /**
    * Functions to add new question to Questionnaire
-   * @param questionText 
-   * @param options 
+   * @param question
+   * @param options
    */
-  const handleAddQuestionSubmit = (
-    questionText: string,
-    options: QuestionOption[]
-  ) => {
-    setQuestions((prevQuestions: Question[]) => [...prevQuestions, { questionText, options }]);
+  const handleAddQuestionSubmit = (question: string, options: QuestionOption[]) => {
+    setOptions((prevQuestions: Question[]) => [...prevQuestions, { question, options }]);
   };
 
   const handleDeleteQuestion = (index: number) => {
-    const updatedQuestions = [...questions];
+    const updatedQuestions = [...options];
     updatedQuestions.splice(index, 1);
-    setQuestions(updatedQuestions);
+    setOptions(updatedQuestions);
+  };
+
+  /**
+   * Function to count all correct answers in the questionnaire, used for passScore determination
+   */
+  const countCorrectAnswers = () => {
+    return options.reduce((count, question) => {
+      return count + question.options.filter((option) => option.value === true).length;
+    }, 0);
   };
 
   /**
-   * This should save the question title and questions to DB.
-   * TODO:
+   * Function to save the new questionnaire
    */
-  const handleSaveSubmit = () => {};
+  const handleSaveSubmit = async () => {
+    try {
+      setLoading(true);
+      setError(null);
+
+      const createdQuestionnaire = await questionnaireApi.createQuestionnaire({
+        questionnaire: {
+          id: "",
+          title: questionnaireTitle,
+          description: questionnaireDescription,
+          options: options,
+          passScore: passScoreValue
+        }
+      });
+      return createdQuestionnaire;
+    } catch (error) {
+      console.error("Failed to save questionnaire", error);
+      setError("Failed to save questionnaire. Please try again.");
+      setTimeout(() => {
+        setError(null);
+      }, 3000);
+    } finally {
+      setLoading(false);
+      setQuestionnaireTitle("");
+      setQuestionnaireDescription("");
+      setOptions([]);
+      setPassScoreValue(0);
```

Comment:
This might require some refactoring if you update the state as I suggest in a previous comment. It might also be worth grouping these into a seperate "closeAndClear" function. Especially if we think such resetting of the state might end up needing to happen in multiple places?

Reply:
Now in useState declarations I define what are values when page mounts. Would it be useful to use useEffect() and this closeAndClear() to set those empty values in page loading or is this setup ok ?